app/src/fdroid/java/org/fossasia/openevent/general/welcome/WelcomeFragment.kt

```diff
@@ -23,7 +22,9 @@ class WelcomeFragment : Fragment() {
         rootView.currentLocation.visibility = View.GONE
 
         rootView.pickCityButton.setOnClickListener {
-            Navigation.findNavController(rootView).navigate(R.id.searchLocationFragment, null, Utils.getAnimSlide())
+            WelcomeFragmentDirections.actionWelcomeToSearch().let {
+                findNavController().navigate(it)
+            }
```

Comment:
Why not `findNavController().navigate(WelcomeFragmentDirections.actionWelcomeToSearch())?`

Reply:
@iamareebjamal For ease of adding variables if added future

Reply:
IDK how will it make it easier. Give an example

Reply:
```
 WelcomeFragmentDirections.actionWelcomeToSearch(query).let {
                findNavController().navigate(it)
            }
```
findNavController().navigate(WelcomeFragmentDirections.actionWelcomeToSearch(
query = "",
            location = Preference().getString(SAVED_LOCATION).toString(),	            location = Preference().getString(SAVED_LOCATION).toString(),
            type = getString(R.string.anything),	            type = getString(R.string.anything),
            date = time)	            date = time


))

Reply:
@iamareebjamal the first one look much nice in terms of viewability although it is just a preference hardly made a difference

Reply:
I think it looks more verbose. There is no use of let here. When you add more arguments to first method, it'll look just as bad

Reply:
I'll refractor it